CMakeLists.txt

```diff
@@ -253,6 +251,13 @@ add_compile_options(/MP) # Build with Multiple Processes (number of processes eq
 add_compile_options(/GS) # Buffer Security Check
 add_compile_options(/Zm200) # Specify Precompiled Header Memory Allocation Limit of 150MB
 add_compile_options(/wd4960 /wd4961 /wd4603 /wd4627 /wd4838 /wd4456 /wd4457 /wd4458 /wd4459 /wd4091 /we4640)
+add_compile_options(/O1) # minimize space
+
+if (CMAKE_BUILD_TYPE STREQUAL Release OR CMAKE_BUILD_TYPE STREQUAL Relwithdebinfo)
+  add_compile_options(/GL) # enable WholeProgramOptimization
```

Comment:
I just realized that the first line of commit log doesn't make sense, as there is no /WPO switched added in the commit. 

You can probably change it like so:

```
[build, win]: Enables Whole Program Optimization.

* This commit enables whole program optimization for release
  configuration on Windows.

* Fix all messages "/LTCG specified but no code generation required;
  remove /LTCG from link command line to improve linker performance"
  which slow down the build process.

* Fix all messages "MSIL .netmodule or module compiled with /GL found;
  restarting link with /LTCG; add /LTCG to the link command line to
  improve linker performance" which slow down the build process.

* The previous two fixes work in Release and Debug builds on Windows.

Issue-URL: #1086
```